File: src/main/java/com/itechart/demojavakotlin/entity/MovieEntity.java
package com.itechart.demojavakotlin.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "movie")
public class MovieEntity {

    @Id()
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id")
    private UUID id;

    @Column(name = "name", nullable = false, unique = true)
    private String name;

    @Column(name = "director", nullable = false)
    private String director;

    @Column(name = "description")
    private String description;

    @Column(name = "tickets_quantity")
    private Integer ticketsQuantity;

    @Column(name = "ticket_price")
    private BigDecimal ticketPrice;

    @JsonIgnore
    @OneToMany(mappedBy = "movieId", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<TicketEntity> tickets;
}

File: kotlinpath/demojavakotlin/src/main/kotlin/by/itechart/demojavakotlin/entity/MovieEntity.kt
package by.itechart.demojavakotlin.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import java.math.BigDecimal
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "movie")
data class MovieEntity(
        @Id
        @Column(name = "id")
        @GeneratedValue
        val id: UUID = UUID.randomUUID(),

        @Column(name = "name", nullable = false, unique = true)
        val name: String,

        @Column(name = "director", nullable = false)
        val director: String,

        @Column(name = "description")
        val description: String? = null,

        @Column(name = "tickets_quantity")
        val ticketsQuantity: Int = 0,

        @Column(name = "ticket_price")
        val ticketPrice: BigDecimal = BigDecimal(0),

        @JsonIgnore
        @OneToMany(mappedBy = "movieId", fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
        val tickets: List<TicketEntity>? = null
)
